catalog/transformer/catalog-transformer-csv-common/src/test/java/ddf/catalog/transformer/csv/common/CsvTransformerTest.java

```diff
@@ -0,0 +1,223 @@
+    {"attribute4", 3.14159, BasicTypes.DOUBLE_TYPE},
+    {"attribute5", "value,5", BasicTypes.STRING_TYPE},
+    {"attribute6", "value6", BasicTypes.STRING_TYPE},
+    {"attribute7", "OBJECT", BasicTypes.OBJECT_TYPE},
+    {"attribute8", "BINARY", BasicTypes.BINARY_TYPE}
+  };
+
+  @Before
+  public void setup() {
+    ATTRIBUTE_DESCRIPTOR_LIST.clear();
+    metacardList.clear();
+    buildMetacardDataMap();
+    buildMetacardList();
+  }
+
+  @Test
+  public void getAllAttributes() {
+    Set<String> hiddenFields = new HashSet<>();
+    hiddenFields.add("attribute1");
+
+    Set<AttributeDescriptor> allAttributes =
+        CsvTransformer.getAllAttributes(metacardList, hiddenFields);
+    assertThat(allAttributes, hasSize(5));
+
+    Set<String> allAttributeNames =
+        allAttributes.stream().map(AttributeDescriptor::getName).collect(Collectors.toSet());
+
+    final Set<String> expectedAttributes =
+        Sets.newHashSet("attribute2", "attribute3", "attribute4", "attribute5", "attribute6");
+
+    assertThat(allAttributeNames, is(expectedAttributes));
+  }
+
+  @Test
+  public void getOnlyRequestedAttributes() {
+    Set<String> requestedAttributes = new HashSet<>();
+    requestedAttributes.add("attribute1");
+
+    Set<AttributeDescriptor> onlyRequestedAttributes =
+        CsvTransformer.getOnlyRequestedAttributes(
+            metacardList, requestedAttributes, Collections.emptySet());
+
+    assertThat(onlyRequestedAttributes, hasSize(1));
+    final String name = onlyRequestedAttributes.stream().findFirst().get().getName();
+    assertThat(name, is("attribute1"));
+  }
+
+  @Test
+  public void getOnlyRequestedAttributesWithExcludedConfig() {
+    Set<String> requestedAttributes = Sets.newHashSet("attribute1", "attribute2");
+    Set<String> excludedAttributes = Sets.newHashSet("attribute1");
+
+    Set<AttributeDescriptor> onlyRequestedAttributes =
+        CsvTransformer.getOnlyRequestedAttributes(
+            metacardList, requestedAttributes, excludedAttributes);
+
+    assertThat(onlyRequestedAttributes, hasSize(1));
+    final String name = onlyRequestedAttributes.stream().findFirst().get().getName();
+    assertThat(name, is("attribute2"));
+  }
+
+  @Test
+  public void writeSearchResultsToCsv() throws CatalogTransformerException {
+    List<AttributeDescriptor> requestedAttributes = new ArrayList<>();
+    requestedAttributes.add(buildAttributeDescriptor("attribute1", BasicTypes.STRING_TYPE));
+
+    Appendable csvText =
+        CsvTransformer.writeMetacardsToCsv(
+            metacardList, requestedAttributes, Collections.emptyMap());
+
+    Scanner scanner = new Scanner(csvText.toString());
+    scanner.useDelimiter(CSV_REGEX);
+
+    // Only attributes in "attributes" config field will exist in output.
+    // OBJECT types, BINARY types and excluded attributes will be filtered out
+    // excluded attribute take precedence over attributes that appear in requested attribute list.
```

Comment:
Everything in these comments should be detailed in the implementation rather than in the test (preferably by intuitive method signatures and code, but also by Javadoc if necessary).